app/views/finance/provider/settings/show.html.slim

```diff
@@ -1,135 +1,114 @@
+                                  collection: Finance::BillingStrategy::CURRENCIES,
+                                  include_blank: false
+
+          = form.semantic_fields_for :account do |account|
+            = account.input :invoice_footnote, as: :patternfly_textarea, input_html: { rows: 2 }
+            = account.input :vat_zero_text, as: :patternfly_textarea, input_html: { rows: 2 }
+
+          = form.input :numbering_period, as: :patternfly_select, collection: ['monthly', 'yearly'],
+                                          include_blank: false
+
+          div class="pf-c-form__group pf-m-action"
+            div class="pf-c-form__actions"
+              button class="pf-c-button pf-m-primary" type="submit" Save
+
+  - if can?(:manage, :finance) && @account.billing_strategy.try(:needs_credit_card?)
+    div class="pf-l-flex__item"
+    div class="pf-c-card"
+      div class="pf-c-card__title" Credit card gateway
+      div class="pf-c-card__body"
+        = semantic_form_for @account, url: admin_account_payment_gateway_path,
+                                      html: { id: 'payment-gateway-form', class: 'pf-c-form pf-m-limit-width' } do |form|
+          = form.input :payment_gateway_type, as: :patternfly_select,
+                                              member_value: :type,
+                                              collection: @payment_gateways,
+                                              include_blank: 'Choose one'
+
+          = form.fields_for :payment_gateway_options do |fields|
+            - @payment_gateways.each do |payment_gateway|
+              - if payment_gateway.fields.present?
+                - is_current = payment_gateway.type == @account.payment_gateway_type
+                section class="pf-c-form__section payment_gateway_options #{is_current ? '' : 'hidden'}" id="payment_gateway_#{payment_gateway.type}" role="group"
+                  - payment_gateway.fields.each do |name, label|
                     - if payment_gateway.boolean_field_keys.include?(name)
-                      = fields.label name, for: ['account', 'payment', payment_gateway.type, name].join('_') do
-                        - if enabled
-                          = fields.check_box name, checked: @account.payment_gateway_options[name].present?, id: ['account', 'payment', payment_gateway.type, name].join('_')
-                        - else
-                          = fields.check_box name, checked: false, disabled: true, id: ['account', 'payment', payment_gateway.type, name].join('_')
-
-                        = label
-                      p.inline-hints Make sure your Braintree account has 3D Secure enabled before you enable it here as otherwise payments initiated from the developer portal will fail.
-
+                      - if is_current
+                        = fields.input name, as: :patternfly_checkbox,
+                                             id: ['account', 'payment', payment_gateway.type, name].join('_'),
```

Comment:
A proper label is missing when current :upside_down_face: 

![image](https://github.com/3scale/porta/assets/1270328/f136a16a-9913-4350-92c0-c60e34e8ed1c)